feat(cli): add core functionality to npx expo lint command (#36262)
expo/expo: docs/pages/more/expo-cli.mdx

Native source code must be generated before a native app can compile. Expo CLI provides a unique and powerful system called _prebuild_, that generates the native code for your project. To learn more, read the [Expo Prebuild docs](/workflow/prebuild/).

## Lint

<Terminal cmd={['$ npx expo lint']} />

> Linting helps enforce best practices and ensure your code is consistent. The `npx expo lint` command will setup ESLint with Expo-specific settings and run the `npx eslint` command with options that are optimized for the Expo framework. You can fix all automatically fixable issues by running `npx expo lint --fix`.

Reviewer: ```suggestion
Linting helps enforce best practices and ensure your code is consistent. The `npx expo lint` command will set up ESLint with Expo-specific settings and run the `npx eslint` command with options that are optimized for the Expo framework. By running `npx expo lint --fix`, linting issues can be fixed automatically.
```

Last sentence wasn't clear.

> Running `npx expo lint` targets all files in the **/src**, **/app**, and **/components** directories by default. You can pass custom files or directories to lint by passing them as arguments, e.g. `npx expo lint ./utils constants.ts`.

Reviewer: ```suggestion
Running `npx expo lint` targets all files in the **/src**, **/app**, and **/components** directories by default. You can also pass custom files or directories to lint by passing them as arguments. For example, `npx expo lint ./utils constants.ts`.
```

The second sentence is an option feature description. Let's emphasize it that way.

> All files matching `.js, .jsx, .ts, .tsx, .mjs, .cjs` extensions will be linted by default, you can customize the extensions by passing the `--ext` flag. For example, to lint only `.ts` and `.tsx` files, you can run: `npx expo lint --ext .ts,.tsx` or `npx expo lint --ext .js --tsx .tsx`.

Reviewer: ```suggestion
All files matching `.js, .jsx, .ts, .tsx, .mjs, .cjs` extensions will be linted by default. You can customize the extensions by passing the `--ext` flag. For example, to lint only `.ts` and `.tsx` files, you can use the `--ext` option: `npx expo lint --ext .ts,.tsx` or `npx expo lint --ext .js --tsx .tsx`.
```

If you need additional customization, you can pass extra arguments using the `--` operator. For example, to pass the `--no-error-on-unmatched-pattern` flag to ESLint, you can run: `npx expo lint -- --no-error-on-unmatched-pattern`. If you need even more customization, you can use `npx eslint` directly.

<BoxLink

>   title="Linting in Expo"

Reviewer: ```suggestion
  title="Using ESLint in Expo"
```
